src/nvim/os/env.c

```diff
@@ -395,9 +395,16 @@ void os_get_hostname(char *hostname, size_t size)
 
 /// To get the "real" home directory:
 ///   - get value of $HOME
+/// For Windows:
+///   - assemble homedir using HOMEDRIVE and HOMEPATH
+///   - try os_homedir()
+///   - resolve references
+///   - guess C drive
 /// For Unix:
+///   - try os_homedir()
 ///   - go to that directory
 ///   - do os_dirname() to get the real name of that directory.
+///   - as a last resort, get the current directory.
 /// This also works with mounts and links.
 /// Don't do this for Windows, it will change the "current dir" for a drive.
```

Comment:
The 2 lines above suggest that `go to that directory` and then `os_dirname()` should not run on Windows.